### PR TITLE
[misc] fix express deprecations

### PR DESCRIPTION
**Pull request by das7pad**
### Description

For https://github.com/overleaf/issues/issues/2965

I paired with @warncke on this.

#### Related Issues / PRs

https://github.com/overleaf/issues/issues/2965

#### Potential Impact

High, but all usage should be covered by unit/acceptance tests.

### CODE REVIEW

File: test/unit/js/HttpControllerTests.js [hunk 1]
```diff
@@ -33,6 +33,7 @@ describe('HttpController', function() {
     this.res.status = sinon.stub().returns(this.res)
     this.res.end = sinon.stub()
     this.res.send = sinon.stub()
```
Comment (mans0954) [inline test/unit/js/HttpControllerTests.js:35]: Do you still need the `send` stub?

**Review by mans0954 — APPROVED** [review]

File: test/unit/js/HttpControllerTests.js (hunk 1 above)
Comment (das7pad) [reply test/unit/js/HttpControllerTests.js:35]: Yes.

<details>

<pre>
  1) HttpController
       addContact
         with an invalid contact id
           "before each" hook for "should return 400, Bad Request":
     TypeError: res.status(...).send is not a function
      at Object.addContact (app/js/HttpController.js:22:23)
      at Context.<anonymous> (test/unit/js/HttpControllerTests.js:71:36)

  2) HttpController
       getContacts
         normally
           "before each" hook for "should look up the contacts in mongo":
     TypeError: res.status(...).send is not a function
      at /app/app/js/HttpController.js:75:30
      at callCallback (node_modules/sinon/lib/sinon/behavior.js:101:25)
      at Object.invoke (node_modules/sinon/lib/sinon/behavior.js:146:27)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:39:43)
      at Function.invoke (node_modules/sinon/lib/sinon/proxy-invoke.js:47:47)
      at Object.functionStub [as getContacts] (node_modules/sinon/lib/sinon/proxy.js:215:26)
      at Object.getContacts (app/js/HttpController.js:55:27)
      at Context.<anonymous> (test/unit/js/HttpControllerTests.js:99:36)

  3) HttpController
       getContacts
         with more contacts than the limit
           "before each" hook for "should return the most commonly used contacts up to the limit":
     TypeError: res.status(...).send is not a function
      at /app/app/js/HttpController.js:75:30
      at callCallback (node_modules/sinon/lib/sinon/behavior.js:101:25)
      at Object.invoke (node_modules/sinon/lib/sinon/behavior.js:146:27)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:39:43)
      at Function.invoke (node_modules/sinon/lib/sinon/proxy-invoke.js:47:47)
      at Object.functionStub [as getContacts] (node_modules/sinon/lib/sinon/proxy.js:215:26)
      at Object.getContacts (app/js/HttpController.js:55:27)
      at Context.<anonymous> (test/unit/js/HttpControllerTests.js:120:36)

  4) HttpController
       getContacts
         without a contact list
           "before each" hook for "should return an empty list":
     TypeError: res.status(...).send is not a function
      at /app/app/js/HttpController.js:75:30
      at callCallback (node_modules/sinon/lib/sinon/behavior.js:101:25)
      at Object.invoke (node_modules/sinon/lib/sinon/behavior.js:146:27)
      at Object.functionStub (node_modules/sinon/lib/sinon/stub.js:39:43)
      at Function.invoke (node_modules/sinon/lib/sinon/proxy-invoke.js:47:47)
      at Object.functionStub [as getContacts] (node_modules/sinon/lib/sinon/proxy.js:215:26)
      at Object.getContacts (app/js/HttpController.js:55:27)
      at Context.<anonymous> (test/unit/js/HttpControllerTests.js:137:36)
</pre>
</details>

**Comment by das7pad** [comment]
Zenhub: Putting this in Held for a deploy on Monday

**Comment by das7pad** [comment]
Force-pushed to fix a conflict with the format fixes #34

**Pull request merged**